src/compiler/utilities.rs

```diff
@@ -162,6 +200,13 @@ impl DiagnosticCollection {
     }
 }
 
+pub fn get_effective_type_annotation_node(node: &Node) -> Option<Rc<Node /*TypeNode*/>> {
+    let type_ = node
+        .maybe_as_has_type()
+        .and_then(|has_type| has_type.type_());
+    type_
```

Comment:
This is another place where you could use `?` and maybe make this a bit more concise.